nodes.go

```diff
@@ -620,10 +621,9 @@ func (s *NodesService) putOrOverwrite(in io.Reader, httpVerb, url, name, metadat
 	contentType := writer.FormDataContentType()
 	contentLength := int64(-1)
 
-	buf := make([]byte, 1)
-	n, err := in.Read(buf)
-	isZeroLength := n == 0 && err == io.EOF
-	in = io.MultiReader(bytes.NewReader(buf), in)
+	bufIn := bufio.NewReader(in)
```

Comment:
I think using a bufio is an interesting idea, but I don't think we should introduce more buffering if we can avoid it.  rclone already implements streaming buffers for big files and this will just slow things down with more data copying.

I think the previous approach was probably better as it will copy the data less.

However in the previous approach you want to use io.ReadFull rather than io.Read which is quite at liberty to return `0, nil`.  I note that I've made the same mistake in the end of read detection in amazonclouddrive.go!

So I think it should be something like this - note it needs to return errors here also.

```go
n, err := io.ReadFull(in, buf)
isZeroLength := err == io.EOF
if !isZeroLength && err != nil {
    return nil, nil, err
}
in = io.MultiReader(bytes.NewReader(buf), in)
```